toolkit/hooks/src/use-date-helper.tsx

```diff
@@ -1,47 +1,76 @@
-import React from "react";
 import { DateTime } from "ts-luxon";
 
-const convertGMTDateToTimezone: React.FC<any> = (
-	date: string | Date,
-	format = "MMM dd, yyyy, hh:mm a",
-) => {
-	if (typeof date === "string") {
-		const dateTime = DateTime.fromISO(date, { zone: "local" }).toFormat(format);
-		return dateTime;
-	} else {
-		const dateTime = DateTime.fromJSDate(date, { zone: "local" }).toFormat(
-			format,
-		);
+function useDateHelpers() {
+	const addLeadingZeroes = (date: string) => {
+		return date.replace(/(\b\d\b)/g, "0$&");
+	};
 
-		console.log(dateTime, "DateTime____");
+	const MonthFormat = ["MM", "M"];
+	const DayFormat = ["dd", "d"];
+	const YearFormat = ["yyyy"];
 
-		return dateTime;
-	}
-};
+	const parseCustomDate = (date: string, format = "dd/MM/yyyy") => {
+		date = addLeadingZeroes(date);
+		const formatParts = format.split(/[- /]/);
+		const dateParts = date.split(/[- /]/);
 
-function useDateHelpers() {
-	const getFormattedDate = (
-		date: string,
-		timestamp = false,
-		format = "MMM dd, yyyy, hh:mm a",
-	) => {
-		console.log("DATE_VALUE", date);
-		if (timestamp) {
-			date = new Date(parseInt(date) * 1000).toISOString();
+		if (formatParts.length !== dateParts.length) {
+			throw new Error("Date format and input date do not match.");
 		}
+		const dateValues: {
+			year: number;
+			month: number;
+			day: number;
+		} = {
+			year: 0,
+			month: 0,
+			day: 0,
+		};
+
+		for (let i = 0; i < formatParts.length; i++) {
+			const formatPart = formatParts[i];
+			const datePart = dateParts[i];
 
-		return convertGMTDateToTimezone(date, format);
+			if (YearFormat.includes(formatPart)) {
+				dateValues.year = parseInt(datePart);
+			} else if (MonthFormat.includes(formatPart)) {
+				dateValues.month = parseInt(datePart);
+			} else if (DayFormat.includes(formatPart)) {
+				dateValues.day = parseInt(datePart);
+			}
+		}
+		if (!dateValues.year || !dateValues.month || !dateValues.day) {
+			throw new Error("Invalid date format or input date.");
+		}
+		return new Date(dateValues.year, dateValues.month - 1, dateValues.day);
 	};
 
-	const isValidDate = (date: string, format = "MM/dd/yy") => {
-		const normalizedDate = date.replace(/(\d)\/(\d)/, "0$1/0$2");
+	const getFormattedDate = (date: Date, format = "dd/MM/yyyy") => {
+		const newDate = DateTime.fromISO(date.toISOString());
+		const n = newDate.toFormat(format);
```

Comment:
Its better to make this variable more discriptive. Will help for future debugging or chnages.
```suggestion
		const formattedDate = newDate.toFormat(format);
```